spec/system/jobseekers_can_view_a_vacancy_spec.rb

```diff
@@ -1,22 +1,34 @@
 require "rails_helper"
 
 RSpec.describe "Viewing a single published vacancy" do
-  let(:school) { create(:school) }
+  let(:school_group) { create(:local_authority) }
+  let(:school1) { create(:school) }
+  let(:school2) { create(:school) }
+  let(:vacancy) do
+    create(:vacancy, vacancy_trait, :at_multiple_schools, :with_supporting_documents, :no_tv_applications,
+           organisation_vacancies_attributes: [
+             { organisation: school1 }, { organisation: school2 }
+           ],
+           working_patterns: %w[full_time part_time],
+           subjects: %w[Physics])
+  end
 
   before do
-    vacancy.organisation_vacancies.create(organisation: school)
+    SchoolGroupMembership.find_or_create_by(school_id: school1.id, school_group_id: school_group.id)
```

Comment:
Why find_or_create_by ?

Reply:
I forget the precise reason, but I think for some tests the membership already existed, so, if it does, we shouldn't create it again.

Reply:
Ah that makes sense